exercises/grade-school/src/example.h

```diff
@@ -0,0 +1,20 @@
+#ifndef GRADE_SCHOOL_H
+#define GRADE_SCHOOL_H
+
+#include <stddef.h>
+#include <stdint.h>
+
+#define MAX_NAME_LENGTH 20
+
+typedef struct {
+   uint8_t grade;
+   char *name;
+} student_t;
+
+char **roster(size_t count, student_t * students);
+
+char **grade(size_t count, student_t * students, uint8_t desired_grade);
+
+void delete_listing(char **listing);
```

Comment:
This interface is pretty confusing to me. Even looking at the tests I'm having a hard time understanding what `roster` and `grade` do. When I've done this exercise in other tracks there's generally a pretty clear flow:

- Create school
- Add students
- Query school
- Cleanup (if necessary)

Do these functions map to this flow?

Reply:
Ah OK, I interpreted the exercise specification JSON pretty literally. The description does convey the flow you mention a bit more.

That said, this was an interesting one to think about what the best way to solve it was in relation to the interface implied by the specification JSON. But on the other hand, it was interesting because it was complex 🤣 
(I realised as well that there is no specification independent way of know the length of the returned arrays 🙈  )

I'll have a go at reworking it with the flow you suggest and see how it looks thanks 👍